### DISCUSSION TITLE
'so-test' command fail

### DISCUSSION THREAD

**Discussion by lconne81** [Security-Onion-Solutions/securityonion#3331] [Unsupported Versions] [answered]
I ran 'sudo so-test' as instructed in the  Security Onion Essentials - Intro to Analysis Tools video (https://www.youtube.com/watch?v=c18D7Dkoz5E&t=108s) and got the message that is shown in the picture below. This was done on version 2.3.30. Is there anything I can do to resolve this issue?

![SO 'so-test' Fail](https://user-images.githubusercontent.com/80169050/110214959-26133b00-7e75-11eb-8678-f2f11310faf6.JPG)

**Comment by cam98mac** [comment, marked as answer]
I am also a beginner so I don't have any great advice for you but I did want to let you know that so-test does work in version 2.3.30 unlike in the previous version of SO. When I was working through the startup set of videos I ran into trouble getting SO to talk with the internet. You may have already done this but does your install talk to the internet? you can check using the command from inside the SO terminal VM of ping 8.8.8.8 (or any other internet routeable ip address, that one is a Google server). I had to set the network connections on the VM to bridged to get SO to work correctly.

**Reply by lconne81** [reply]
I tried pinging 8.8.8.8 just now and I'm getting "Destination Host Unreachable".  Is there something I have to do during the startup to get SO to talk to the internet?

**Reply by cam98mac** [reply]
Did you set up security onion in a VM or install it directly onto a computer? If on a VM there are two things I would check. When you installed SO you had to specify an IP address for it - the IP address should be on that is compatible with your network - not the one used in the example video. The other as I mentioned above is a problem with the VM itself - I had to set the network connection type to "bridged". I was using VMWare Desktop and not Fusion as was done on the example video. To successfully change the network type you have to shut down the VM.

**Reply by lconne81** [reply]
Yes, I installed it on a VM, using the current ISO image on VirtualBox. The IP address I entered was the IP address for my current location. On the HOME_NET step of the installation, I left it as it was on default.  Adapter 1 is set to "NAT" while Adapter 2 is set to "Internal Network". Which adapter should be switched to bridged? And if my previous installation choices were incorrect, how can I change them without having to reinstall SO in its entirety?

**Reply by cam98mac** [reply]
I don't have experience with VirtualBox, but for me I set both network adapters to Bridged - and that is done in the VM software. I also left the HOME_NET at the default. For me, I had them both on NAT and it wouldn't work. I suspect if you change them both to bridged your installation would work.

**Reply by lconne81** [reply]
I set both adapters to Bridged just now and got the same result as before. I feel like it could have something to do with the replay functionality, but as of now I can't find out if that's the case.

**Reply by cam98mac** [reply]
I'm pretty sure the replay function has to reach out to the internet to download some stuff. But perhaps you are correct. I am not familiar with VirtualBox enough to provide more advice on how to get a VM inside it to talk with the outside world. You could also test if you can get to the SO console website to see if the VM can at least get to the host OS.

**Reply by lconne81** [reply]
Actually, the so-test command works now. I guess I had to give VirtualBox a bit of time to process the change. Thanks for your help @cam98mac!

**Reply by cam98mac** [reply]
You are welcome! I'm glad it is working for you.
